### ISSUE TITLE
No federation back off on certificate error. 

### ISSUE THREAD

**Issue by HenrikSolver** [matrix-org/dendrite#1427]
**Describe the bug**
When the other HS is configured with a faulty certificate the log fills up with lines like.
```
WARN[2020-09-17T15:48:23.670800134Z] [github.com/matrix-org/gomatrixserverlib@v0.0.0-20200907151926-38f437f2b2a6/client.go:140] RoundTrip
         Error sending request to https://steiger.online:8448/_matrix/federation/v1/user/devices/@marco:steiger.online: x509: certificate is valid for f817a0aebc14c45cf8e11479cd5a4d7e.2625b601f91d0765fb05aa13a6cf86a1.traefik.default, not steiger.online  context=missing out.req.ID=v4XLeT9jqWvU out.req.method=GET out.req.uri="matrix://steiger.online/_matrix/federation/v1/user/devices/@marco:steiger.online"
WARN[2020-09-17T15:48:23.671583514Z] [github.com/matrix-org/gomatrixserverlib@v0.0.0-20200907151926-38f437f2b2a6/client.go:389] DoHTTPRequest
         Outgoing request failed                       context=missing error="Get \"matrix://steiger.online/_matrix/federation/v1/user/devices/@marco:steiger.online\": x509: certificate is valid for f817a0aebc14c45cf8e11479cd5a4d7e.2625b601f91d0765fb05aa13a6cf86a1.traefik.default, not steiger.online" out.req.ID=v4XLeT9jqWvU out.req.method=GET out.req.uri="matrix://steiger.online/_matrix/federation/v1/user/devices/@marco:steiger.online"
ERRO[2020-09-17T15:48:23.672394461Z] [github.com/matrix-org/dendrite/keyserver/internal/device_list_update.go:342] processServer
         failed to query device keys for user          context=missing error="Get \"matrix://steiger.online/_matrix/federation/v1/user/devices/@marco:steiger.online\": x509: certificate is valid for f817a0aebc14c45cf8e11479cd5a4d7e.2625b601f91d0765fb05aa13a6cf86a1.traefik.default, not steiger.online - (retry_after=999.994081ms, blacklisted=false)" server_name=steiger.online user_id="@marco:steiger.online"
```

**Expected behavior**
Local  HS should back off, or maybe such a broken installation should not be allowed join the Matrix network in any way.

**Additional context**
Seen on a build from 18231f25b437d2f03b3be1e0536fc46d45c8691f

**Comment by neilalexander** [comment]
This is device list syncing again - I have a similar problem with `aria-net.org` on my HS.

**Comment by wirke** [comment]
Same https://github.com/matrix-org/dendrite/issues/1418

**Comment by neilalexander** [comment]
This should be fixed by #1431.
